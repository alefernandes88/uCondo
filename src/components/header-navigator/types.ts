import type { NavigationProp } from '@react-navigation/native'
import type { IconProps } from '../../icons/types'

export enum ScreenName {
  HOME = 'Home',
  CREATE_ACCOUNT = 'CreateAccount'
}

export interface RootStackParamList {
  [ScreenName.HOME]: undefined
  [ScreenName.CREATE_ACCOUNT]: { accountCode?: string }
};

export type AppNavigationProps = NavigationProp<RootStackParamList>

export interface ActionButtonProps {
  navigation: AppNavigationProps
  path?: ScreenName
  icon: (props: IconProps) => JSX.Element
  onPress?: () => void
}
