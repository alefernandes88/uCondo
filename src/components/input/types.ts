import type { KeyboardType } from 'react-native'

export interface SelectItemProps<T> {
  label: string
  value: T
}

interface InputBase<T> {
  label: string
  value: T
  onValueChange: (value: T) => void
  isDisabled?: boolean
}

export interface SelectProps<T> extends InputBase<T> {
  options: Array<SelectItemProps<T>>
}

export interface InputProps<T = string> extends InputBase<T> {
  placeholder?: string
  keyboardType?: KeyboardType
}
