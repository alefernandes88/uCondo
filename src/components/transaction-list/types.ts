import type { Account } from '../../types/account'

interface TransactionCallbacks {
  onErase: (account: Account) => void
  onSelect: (account: Account) => void
}

export interface TransactionListItemProps extends TransactionCallbacks {
  account: Account
}

export interface TransactionListProps extends TransactionCallbacks {
  data: Account[]
}
