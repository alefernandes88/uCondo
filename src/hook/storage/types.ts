import type { Account } from '../../types/account'

export interface useAllAccountsResponse {
  accounts: Account[]
  refetch: () => void
}
