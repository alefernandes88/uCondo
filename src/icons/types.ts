export interface BaseIcon {
  height: number
  width: number
  color: string
  strokeSize: number
}

export type IconProps = Partial<BaseIcon>
