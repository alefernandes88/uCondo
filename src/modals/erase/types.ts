import type { Account } from '../../types/account'

export interface EraseModalProps {
  isOpen: boolean
  onClose: () => void
  onErase: (account: Account) => void
  account: Account | null
}

export type ButtonTypes = 'primary' | 'secondary'
