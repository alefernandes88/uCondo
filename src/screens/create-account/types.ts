import type { AppNavigationProps } from '../../components/header-navigator/types'
import type { RouteProp } from '@react-navigation/native'

export interface CreateAccountScreenProps {
  navigation: AppNavigationProps
  route: RouteProp<{ params: { accountCode?: string } }, 'params'>
}
