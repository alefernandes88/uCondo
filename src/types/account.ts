export type AccountTransactionType = 'expenses' | 'revenue'

export interface Account {
  parentAccount: string
  code: string
  name: string
  transactionType: AccountTransactionType
  hasIncoming: boolean
}
