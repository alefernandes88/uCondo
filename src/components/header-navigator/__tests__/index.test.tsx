import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import { ActionButton } from '../index'
import ConfirmIcon from '../../../icons/confirm'
import { ScreenName } from '../types'

describe('Components -> Header Navigator', () => {
  const navigateSpy = jest.fn()
  const onPressSpy = jest.fn()
  const pathMock = ScreenName.HOME

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render with a icon', () => {
    const { toJSON } = render((
      <ActionButton
        navigation={{
          navigate: navigateSpy
        } as any}
        icon={ConfirmIcon}
        onPress={onPressSpy}
        path={pathMock}
      />
    ))

    expect(toJSON()).toMatchSnapshot()
  })

  it('should trigger action when pressed', async () => {
    const navigateSpy = jest.fn()
    const onPressSpy = jest.fn()
    const pathMock = ScreenName.HOME

    const { findByTestId } = render((
      <ActionButton
        navigation={{
          navigate: navigateSpy
        } as any}
        icon={ConfirmIcon}
        onPress={onPressSpy}
        path={pathMock}
      />
    ))

    const button = await findByTestId('header-action-button')
    fireEvent.press(button)

    expect(navigateSpy).toBeCalledWith(pathMock)
    expect(onPressSpy).toBeCalled()
  })
})
