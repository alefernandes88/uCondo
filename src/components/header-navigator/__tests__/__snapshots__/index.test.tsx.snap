// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Header Navigator should render with a icon 1`] = `
<View
  style={
    {
      "borderBottomLeftRadius": 50,
      "borderBottomRightRadius": 50,
      "borderTopLeftRadius": 50,
      "borderTopRightRadius": 50,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
    testID="header-action-button"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 20,
          },
        ]
      }
      vbHeight={16}
      vbWidth={20}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M19.375 0.453125H17.7367C17.507 0.453125 17.2891 0.558594 17.1484 0.739062L7.48517 12.9805L2.85157 7.10938C2.78148 7.02038 2.69214 6.94842 2.59025 6.8989C2.48836 6.84937 2.37658 6.82357 2.26329 6.82344H0.625013C0.467982 6.82344 0.381263 7.00391 0.477357 7.12578L6.89689 15.2586C7.19689 15.6383 7.77345 15.6383 8.07579 15.2586L19.5227 0.753125C19.6188 0.633594 19.532 0.453125 19.375 0.453125Z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
